edx_lint/pylint/filters_docstring/filters_docstring_check.py

```diff
@@ -0,0 +1,142 @@
+        DOCSTRING_MISSING_PURPOSE_OR_BADLY_FORMATTED,
+        DOCSTRING_MISSING_OR_INCORRECT_TYPE,
+        DOCSTRING_MISSING_TRIGGER_OR_BADLY_FORMATTED,
+    )
+    def visit_classdef(self, node):
+        """
+        Visit a class definition and check its docstring.
+
+        If the class is a subclass of OpenEdxPublicFilter, check the format of its docstring. Skip the
+        OpenEdxPublicFilter class itself.
+
+        """
+        if not node.is_subtype_of("openedx_filters.tooling.OpenEdxPublicFilter") or node.name == "OpenEdxPublicFilter":
+            return
+
+        docstring = node.doc_node.value if node.doc_node else ""
+        if not (error_messages := self._check_docstring_format(node, docstring)):
+            return
+        for error_message in error_messages:
+            self.add_message(error_message, node=node, args=(node.name,))
+
+    def _check_docstring_format(self, node, docstring):
+        """
+        Check the format of the docstring for errors and return a list of error messages.
+
+        The docstring should have the following structure:
+        1. Description: Any non-empty text followed by a blank line.
+        2. Filter Type: A line that starts with "Filter Type:".
+        3. Trigger: A line that starts with "Trigger:".
+
+        For example:
+
+        ```
+        Purpose:
+        Filter used to modify the certificate rendering process.
+
+        ... (more description)
+
+        Filter Type:
+            org.openedx.learning.certificate.render.started.v1
+
+        Trigger:
+            - Repository: openedx/edx-platform
+            - Path: lms/djangoapps/certificates/views/webview.py
+            - Function or Method: render_html_view
+        ```
+        """
+        error_messages = []
+        if error_message := self._check_purpose_missing_or_badly_formatted(docstring):
+            error_messages.append(error_message)
+        if error_message := self._check_filter_type_missing_or_incorrect(node, docstring):
+            error_messages.append(error_message)
+        if error_message := self._check_trigger_missing_or_badly_formatted(docstring):
+            error_messages.append(error_message)
+        return error_messages
+
+    def _check_purpose_missing_or_badly_formatted(self, docstring):
+        """
+        Check if the purpose is missing or badly formatted.
+
+        If the purpose is missing or badly formatted, return the error message. Otherwise, return.
+        """
+        if not re.search(r"Purpose:\s*.*\n", docstring):
+            return self.DOCSTRING_MISSING_PURPOSE_OR_BADLY_FORMATTED
+        return None
+
+    def _check_filter_type_missing_or_incorrect(self, node, docstring):
+        """
+        Check if the filter type is missing or incorrect.
+
+        If the filter type is missing or incorrect, return the error message. Otherwise, return.
+        """
+        filter_type = node.locals["filter_type"][0].statement().value.value
```

Comment:
note to self: I think I should do some error handling here.

Reply:
I think that makes sense, otherwise I'm 👍 on this

Reply:
I implemented our suggestion here: https://github.com/openedx/edx-lint/pull/480/commits/7455a5e22ad898986db355944890e3a664664c64. Thank you!